core/src/main/java/com/datastrato/gravitino/Configs.java

```diff
@@ -143,13 +161,45 @@ public interface Configs {
           .longConf()
           .createWithDefault(2000L);
 
-  ConfigEntry<Long> KV_DELETE_AFTER_TIME =
+  @Deprecated
+  ConfigEntry<Optional<Long>> KV_DELETE_AFTER_TIME =
       new ConfigBuilder(KV_DELETE_AFTER_TIME_KEY)
           .doc(
-              "The maximum time in milliseconds that the deleted data and old version data is kept")
+              String.format(
+                  "The maximum time in milliseconds that the deleted data and old version data is kept, "
+                      + "max delete time allow is %s ms(30 days), "
+                      + "min delete time allow is %s ms(10 minutes)",
+                  MAX_DELETE_TIME_ALLOW, MIN_DELETE_TIME_ALLOW))
           .version(ConfigConstants.VERSION_0_3_0)
           .longConf()
-          .createWithDefault(DEFAULT_KV_DELETE_AFTER_TIME);
+          .checkRange(MIN_DELETE_TIME_ALLOW, MAX_DELETE_TIME_ALLOW)
+          .createWithOptional();
+
+  ConfigEntry<Long> STORE_DELETE_AFTER_TIME =
+      new ConfigBuilder(STORE_DELETE_AFTER_TIME_KEY)
+          .doc(
+              String.format(
+                  "The maximum time in milliseconds that the deleted data and old version data is kept, "
+                      + "max delete time allow is %s ms(30 days), "
+                      + "min delete time allow is %s ms(10 minutes)",
+                  MAX_DELETE_TIME_ALLOW, MIN_DELETE_TIME_ALLOW))
+          .version(ConfigConstants.VERSION_0_3_0)
+          .longConf()
+          .checkRange(MIN_DELETE_TIME_ALLOW, MAX_DELETE_TIME_ALLOW)
+          .createWithDefault(DEFAULT_STORE_DELETE_AFTER_TIME);
+
+  ConfigEntry<Long> VERSION_RETENTION_COUNT =
+      new ConfigBuilder(VERSION_RETENTION_COUNT_KEY)
+          .doc(
+              String.format(
+                  "The count of versions allowed to be retained, including the current version, "
+                      + "max version retention count is %s, "
+                      + "min version retention count is %s",
+                  MAX_VERSION_RETENTION_COUNT, MIN_VERSION_RETENTION_COUNT))
+          .version(ConfigConstants.VERSION_0_3_0)
```

Comment:
The initial version should be 0.5.0.

Reply:
And the functionality has not implemened, am I right?

Reply:
> And the functionality has not implemened, am I right?

YES.